### PR TITLE
[frontend] sellers can add / modify / view their products

### PR DESCRIPTION
**Pull request by KatoToyoki**
Resolveeeeeeeees part of #94

### (B) ViewShop TYK :x:
- [x] /sellerID/shop
    - just [check here](http://localhost:8080/sellerID/shop)
- [x] /seller/manageProducts
    - just [check here](http://localhost:8080/user/seller/manageProducts)
- [x] /seller/manageProducts/new
    - [check here](http://localhost:8080/user/seller/manageProducts/new) 
    - you can test: 
        - if input NaN to price/stock 
        - if not upload img
        - if no tag
        - quickly add/delete tags then submit
        - delete product
        - input tag
        - delete tag
- [x] /seller/manageProducts/productID
    -  [get here](http://localhost:8080/user/seller/manageProducts) them click the item you just create then edit it
        - if input NaN to price/stock 
        - if not upload img
        - if no tag
        - quickly add/delete tags then submit
        - delete product
        - input tag
        - delete tag

### (D) Product TYK ❌
- [x] /product
    -  please create new product then go to [here](http://localhost:8080/sellerID/shop), find the item you just added, click "more" button

### Testing environment
open 3 terminals
one for `docker-compose up postgres minio `
one for `go run .`
one for `cd frontend` -> `npm i` -> `npm run build`
then open the browser then enter http://localhost:8080/

### examples
whether you upload the square / vertical-quad / horizontal-quad images, it won't affect component's size

- http://localhost:8080/user/seller/manageProducts
image

- http://localhost:8080/user/seller/manageProducts/31
image

- http://localhost:8080/sellerID/shop/32
image

### CODE REVIEW

File: frontend/src/lib/Status.tsx [hunk 1]
```diff
@@ -10,6 +10,9 @@ export const RouteOnNotOK = async (response: Response, navigate: NavigateFunctio
     case 404:
       navigate('/notFound');
       break;
+    case 500:
```
Comment (NOOBDY) [inline frontend/src/lib/Status.tsx:13]: why would `500` navigate to `/notFound`

File: frontend/src/main.tsx [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+import { MutationCache, QueryClient, QueryClientProvider } from '@tanstack/react-query';
 import AuthProvider from '@components/AuthProvider.tsx';
 
-const queryClient = new QueryClient();
+const queryClient = new QueryClient({
+  mutationCache: new MutationCache({
+    onError: (error: Error) => {
+      console.error(error);
+    },
+  }),
```
Comment (NOOBDY) [inline frontend/src/main.tsx]: why is this needed? if you need error handling you might need to create your own [`<ErrorBoundary>`](https://reactrouter.com/en/main/route/error-element)

File: frontend/src/pages/discover/[goodsID].tsx [hunk 3]
```diff
@@ -2,145 +2,129 @@ import '@style/global.css';
...
+        },
+      });
+      if (!response.ok) {
+        RouteOnNotOK(response, navigate);
+      }
+      return (await response.json()) as GetResponseProps;
+    },
+  });
+
+  if (status != 'success') {
```
Comment (NOOBDY) [inline frontend/src/pages/discover/[goodsID].tsx]:
```suggestion
  if (status !== 'success') {
```

**Review by NOOBDY — CHANGES_REQUESTED** [review]

File: frontend/src/lib/Status.tsx (hunk 1 above)
Comment (KatoToyoki) [reply frontend/src/lib/Status.tsx:13]: image
because this error is not right

**Review by Error0229 — DISMISSED** [review]
WOMM

**Comment by x06lan** [comment]
1. I get two alert when change the product price to 111.1  
screenshot-2024-01-06-22-34-01
screenshot-2024-01-06-22-34-50

2. http://localhost:8080/sellerID/shop/32
view shop button no working

**Review by NOOBDY — DISMISSED** [review]

**Review by NOOBDY — APPROVED** [review]

**Review by Error0229 — APPROVED** [review]
